### PR TITLE
Add BT_CORE_SPEC_5_4 = 13 to HCIVersion Enum (BugFix)

### PR DESCRIPTION
**Pull request by hanhsuan**
## Description

The BT 5.4 is out and DUTs is starting to test this spec.
Therefore, following the [previous one](https://github.com/canonical/checkbox/commit/f53f59ac5769dfb4df406715388bc47fcb6da2b0) to add this spec into support list.

## Resolved issues

#834 

## Tests

[non-modified submission](https://certification.canonical.com/hardware/202310-32266/submission/343109/test-results/fail/)
[modified submission](https://certification.canonical.com/hardware/202310-32266/submission/344688/test-results/pass/)

### CODE REVIEW

File: checkbox-support/checkbox_support/vendor/beacontools/scanner.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ class HCIVersion(IntEnum):
     BT_CORE_SPEC_5_1 = 10
     BT_CORE_SPEC_5_2 = 11
     BT_CORE_SPEC_5_3 = 12
+    BT_CORE_SPEC_5_4 = 13
```
Comment (kissiel) [inline checkbox-support/checkbox_support/vendor/beacontools/scanner.py:49]: Could you please help me find a place where this enumeration is used?

**Review by kissiel — COMMENTED** [review]
Question below

File: checkbox-support/checkbox_support/vendor/beacontools/scanner.py (hunk 1 above)
Comment (hanhsuan) [reply checkbox-support/checkbox_support/vendor/beacontools/scanner.py:49]: This is used in `def set_scan_parameters` of `scanner.py` to set different parameters.

**Review by yphus — APPROVED** [review]
I had to do a similar [fix](https://github.com/canonical/checkbox/commit/f53f59ac5769dfb4df406715388bc47fcb6da2b0) in 2022
Thanks for keeping this enum up to date! +1

File: checkbox-support/checkbox_support/vendor/beacontools/scanner.py (hunk 1 above)
Comment (kissiel) [reply checkbox-support/checkbox_support/vendor/beacontools/scanner.py:49]: it's an inverse mapping, this is why me grepping didn't show anything... Now I get it, thank you.

**Review by kissiel — APPROVED** [review]
TYVM!

File: checkbox-support/checkbox_support/vendor/beacontools/scanner.py (hunk 1 above)
Comment (yphus) [reply checkbox-support/checkbox_support/vendor/beacontools/scanner.py:49]: I'm marking this conversation as resolved then

**Pull request merged**